cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_nodegroup_test.go

```diff
@@ -125,12 +126,34 @@ func TestNodeGroupNewNodeGroup(t *testing.T) {
 			return
 		}
 
-		if ng.Name() != tc.name {
-			t.Errorf("expected %q, got %q", tc.name, ng.Name())
+		if ng == nil {
+			t.Fatal("expected nodegroup to be non-nil")
+		}
+
+		var expectedName string
+		var expectedNamespace string
+
+		switch v := (ng.scalableResource).(type) {
+		case *machineSetScalableResource:
+			expectedName = testConfig.spec.machineSetName
+			expectedNamespace = testConfig.spec.namespace
+
+		case *machineDeploymentScalableResource:
+			expectedName = testConfig.spec.machineDeploymentName
+			expectedNamespace = testConfig.spec.namespace
+		default:
+			t.Fatalf("unexpected type: %T", v)
+		}
+
+		expectedID := path.Join(expectedNamespace, expectedName)
+		expectedDebug := fmt.Sprintf("%s (min: %d, max: %d, replicas: %d)", expectedID, tc.minSize, tc.maxSize, tc.replicas)
```

Comment:
The same here about reusing the same expression at multiple places. Rather to have one place which builds a string and be sure that if two strings are different it's caused by different values rather than a type in a format string.